app/WebParsing/PostParser.hs

```diff
@@ -56,44 +52,36 @@ parHtmlToLines tags =
             | otherwise = t:ts
 
 -- Helpers
-categoryParser :: [Tag T.Text] -> T.Text -> Maybe T.Text -> [T.Text] -> SqlPersistM ()
-categoryParser tags fullPostName firstCourse listPartitions = do
-    case parsed of
-        Right (post, categories) -> do
+categoryParser :: [Tag T.Text] -> [[T.Text]] -> T.Text -> Maybe T.Text -> SqlPersistM ()
+categoryParser tags requirements fullPostName firstCourse = do
+    let categories = map parseRequirement requirements
+    let parsedPost = P.parse (postInfoParser fullPostName firstCourse) failedString (innerText tags)
```

Comment:
Just to be clear, the goal should be to remove the `tags` parameter from this function.

If you're looking to ease the transition, instead of doing `innerText tags` you could do something like `T.intercalate "\n" $ flatten requirements`.